src/Console/BaseCommand.php

```diff
@@ -273,4 +289,26 @@ public function executeCommand(CommandInterface|string $command, array $args = [
             return $e->getCode();
         }
     }
+
+    /**
+     * Get the global event manager.
+     *
+     * @return \Cake\Event\EventManagerInterface
+     */
+    public function getEventManager(): EventManagerInterface
+    {
+        return EventManager::instance();
+    }
+
+    /**
+     * Set the application's event manager.
+     *
+     * @param \Cake\Event\EventManagerInterface $eventManager The event manager to set.
+     * @return $this
+     * @throws \InvalidArgumentException
+     */
+    public function setEventManager(EventManagerInterface $eventManager)
+    {
+        throw new InvalidArgumentException('You are not allowed to overwrite the global event manager.');
```

Comment:
This feels weird to me. Implementing the interface is just namesake if we are disallowing the setter and the getter also just wraps the global one.

Reply:
My thinking was, that this would be the first occurence where the global event manager instance could be overwritten via another object (without calling `EventManager::instance($eventManager)` itself of course).

Reply:
I am obviously not suggesting overwriting the global instance here 😆.

Reply:
If we don't overwrite the `setEventManager` we would allow setting an event manager instance in the command object via the `EventDispatcherTrait` which doesn't do anything.
Therefore, I still think throwing an exception is the most user forward thing we can do.

Reply:
I think the issue here is commands shouldn't need to access the global event manager directly. The `CommandRunner` already has an event manager instance (which it gets from the application), so the same one should be used by commands.

So either that instance should be passed to the command constructor or use `BaseCommand::setEventManager()`. Then you won't need to override either the getter or the setter method in the command class and just use what the `EventManagerTratit` provides.

Reply:
Models and Controllers have their own event manager instances. Why commands need to use the global one?

Reply:
Didn't think of that 🤔 
Thanks @ADmad 👍🏻